### ISSUE TITLE
LRC-7 - Blockswap Economics and Reward Structure 

### ISSUE THREAD

**Issue by Haafingar** [loki-project/loki-improvement-proposals#24]
Hey everyone,

It’s that time again — a review of the Loki block reward, and how we can shape it to bring the best outcomes to the Loki Network and its products. 

This LRC is primarily about the proposition to support a new product on the Loki blockchain, Blockswap. This comes at an interesting time for Loki, as we move rapidly towards the PoS Pulse upgrade, which will drop the mining reward entirely — something agreed upon when we last discussed block reward restructures with you. I’ve been doing lots of modelling to try and come up with a design for the Loki block reward that can support Blockswap, while still granting the network the same level of Sybil resistance for Session and Lokinet. 

Adding a second node layer to the network is a really interesting challenge. As discussed last time, we’ve got even more data on realistic staking equilibriums, and we can now more accurately predict the lockup ratios of the existing Service Node network. Adding this second layer, if incentivized correctly, could send the lockup ratio of Loki far beyond the 45% ratio we have now. Even if that new lockup is being used to secure collateral in Blockswap, it still isn’t liquid — so it can’t be used in a Sybil attack on the network. So the additional lockup brought about by the introduction of Blockswap should result in increased protection for the entire network.

**The Requirements**

Vault nodes must be collateralized sufficiently to prevent ransom attacks and outward theft by a supermajority. Intuitively, one might expect that to adequately protect $1m of liquidity, vault nodes would need significantly more than $1m of collateral to protect that liquidity from being stolen by a malicious attacker with enough vault nodes to form a supermajority. In this instance, if the attacker was able to run $660k worth of nodes in a network with a total collateralization of $1m, they could form a supermajority and steal the vault’s $1m without penalty. 

With that in mind, you might be led to believe that Blockswap will need heavy overcollateralization to be considered secure. But in fact, there are some clever design tricks we can use to secure vaults of an equal or larger value than the collateral associated with the vault nodes operating them. The two methods we’ll employ to achieve this are:

- **Vault splitting**, which involves deterministically shuffling members of a given vault and splitting them up so each vault node has custody of as few vaults simultaneously as possible, but in such a way that the system cannot be influenced to drive a given actor’s nodes into one particular vault. This way, even if the attacker controls two thirds of the nodes, it is very unlikely that they would be able to form a supermajority in any one vault. By slowing down the intervals of vault reshuffles, the attacker must wait an impractically long time before random chance will work in their favour. And even if this did occur, it’s likely they would only gain a supermajority in one vault, so the other vaults would remain protected from that theft.

- **Slashing**, which involves any instance of theft from the vaults (ie. unexpected transactions) to cause the offending vault nodes to have their Loki collateral deducted from their wallet, effectively neutralising the benefit of stealing any amount of cryptocurrency worth the same or less than that of the collateral they have already gained.

To illustrate this, let’s look at a scenario where there are 100 active vault nodes, 90 of which are participating in vaults, and where there are 3 vaults with the following balances:

Loki: $2m (30 nodes)
Eth: $1m (30 nodes)
Btc: $1m (30 nodes)
10 active but out of rotation

In this scenario, the attacker must own at least 20 nodes to have any chance of a successful attack. However, the chances that all of the attacker’s nodes will be randomly selected to become a part of the same vault is practically zero. We ran simulations on the effectiveness of vault splitting, using a probability calculation called a hypergeometric distribution, and [here’s some R code and some results](https://github.com/Haafingar/simon/blob/master/vault%20splitting). Because Blockswap vault iterations only occur every 2-4 weeks, the amount of time an attacker would need to pull off a theft that overcomes vault splitting is substantially higher than if all vaults were run by the same group of nodes. 

If that attacker owns 33 nodes, they'd have a chance of 0.02% every year on average of forming a supermajority in a vault. Looking at a more dangerous scenario, if the attacker had 45 nodes, or just under half the network (which as I’ll show later, is going to cost *a lot*), and there were three active vaults which rotated at a period of 4 weeks, the chance that they’d achieve control over a single vault is a 15.1% chance over a year. However, in this instance, they only control one of the 3 vaults, at most $2m, but more likely $1m. Still, this could mean the attacker could walk away with $1-2m if they owned half the network and were extremely patient. 

This is where slashing comes in. Because of the nature of the quoting system, any unexpected transaction leaving the vault (i.e. a theft) would be noticeable to all other network participants. Seeing as the offending nodes in the vault must sign this transaction, depending on the vault type, they can be identified immediately, or if not, the entire vault can be considered bad. If an offending transaction is created, the Service Node network can slash the collateral of all of those nodes. This means that a thief must also factor in the cost of the collateral they have used in the attack. In the case where a whole vault gets slashed, even if there are some honest nodes in there, this may seem inequitable. However, because of this penalty in place, the attack should never occur.

In this scenario, the attacker gets lucky and controls 50 nodes and randomly wins a supermajority of 20 in the Loki ($2m) pool. Instead of being able to control the entire $4m of liquidity in the system, the attacker only gains control over $2m. If the stake of the 20 nodes is worth as much or more than $2m, the attack is pointless, as the collateral they have will be slashed. It becomes even more pointless considering that the remaining collateral they have and the Loki they have just stolen will now likely be worth less as the market panics. If they gain control of a smaller vault instead of the Loki vault, the attack is even less viable.

By splitting the vaults and slashing bad actors, Blockswap can rely on vault nodes even when they are not substantially overcollateralized. There is no hard limit to what would be considered an ‘acceptable’ collateralization — the greater the collateralization, the less likely an attack of this type is possible. But given the extreme costs of owning large percentages of the circulating supply, the low iteration speed on vault rotations, the splitting of vaults into randomly selected groups of nodes, and the slashing of collateral upon a theft, an attack of this kind would not be viable. That is, unless the value of the liquidity of the nodes exceeds that of the total value of all of the vault nodes in operation. That being said, the total value could exceed the collateralization to a moderate extent without greatly impacting the security of the system. Thus, I argue that we should target Blockswap vault nodes to be collateralized for as much liquidity as we intend Blockswap to hold.

**Target Liquidity**

UniSwap holds an extraordinary $150m in liquidity, but for most of its history has held much less than that, at around the $30m mark. This is a very high amount of liquidity, but considering it is able to capture the entire ERC20/ETH market, this is no real surprise.

In designing the economics for Blockswap, we must decide how much liquidity we want it to support. The more liquidity, the lower the fees for traders, and the greater volume of trades will be passed through the system. More trades will result in more Loki burned in fees, and greater rewards for liquidity providers.

In setting an initial liquidity target, we are not defining the ultimate hard limit of what is possible on Blockswap, but what an initial target is for the first steps in decentralized cross-chain swaps. The number we select is fairly arbitrary, because of course, the more the better. I believe that setting an initial liquidity target of $7 million is sufficient for the purposes of initializing this product.

Granted, this target is sufficiently less than what is currently available on UniSwap. However, this target can be revised over time, and as I’ll explore, it isn’t a good idea to simply ramp up the target liquidity to compete directly with the size of liquidity Uniswap has reached after 2 years of development and widespread adoption.

**Creating the Incentive**

To convince node operators to lock up 7 million dollars worth of Loki, we must create an incentive structure that sufficiently rewards node operators for locking up their capital and behaving correctly on the network.

The reality is that the incentive for vault nodes comes at a network cost in the form of emission. Given that this cost is realised programmatically and not in accordance with how much liquidity is actually in the system (at least initially), we must limit the amount of rewards given to vault nodes. As discovered in our last exploration of this topic, the velocity of money is dramatically reduced with the introduction of staking, so the emission created amongst the vault nodes is not something that is going to be immediately paid for on the market. However emission does have an effect on the overall value of each token on the network over extended time frames, so we should avoid adding emissions wherever possible. 

That being said, if Blockswap is a successful product, there will be lots of Loki being used and burned in each swap. If done in sufficiently large numbers, this will be able to offset the newly created coins awarded to the vault node operators.

After running the Loki Service Node network for nearly 2 years, we now have lots of data on how real world node operators behave. We’ve written about, modelled, and predicted the likely behaviour of Service Node operators and their staking habits, and now have a really good data set to use in modelling for vault nodes.

One of the key concepts we explored in our last discussion was that of the staking equilibrium. That is, what is the minimum amount of returns available to stakers before there are as many people unstaking as there are new entrants staking into the network? I predicted that over time the staking equilibrium would in fact decrease. As Loki becomes a more stable and mature product, I had assumed people would be more likely to tolerate a lower amount of returns, and risk locking up their coins in staked nodes in order to earn returns on a more stable ecosystem.

What seems to be the case is that the staking equilibrium is slightly higher than I predicted, and I believe the reason for that is because of the relatively low Service Node staking requirement.  When the Service Node network first launched, Service Node operators required a whopping 45,000 Loki in order to start a new Service Node. This was done to ensure that the cost of operating a node would not discourage people from seeking those returns. However, as the staking requirement has decreased towards 15,000 Loki, it seems that because of the relatively low collateral amount, and the relative difficulty of operating a large number of Service Nodes, larger Loki holders have found it difficult to maintain a stake in the Loki ecosystem.

Blockswap could be a great solution to this problem, as node operators will now have two options - one being a relatively inexpensive node that operates Proof-of-Stake and the other Loki Services, and another class of node that has a higher operating cost, but allows for lots of Loki to be staked in one transaction. This means for larger holders there is less hassle involved with operating nodes, and an entirely new subset of emission to compete to earn rewards from.

I therefore estimate that although our current staking equilibrium seems to be about 20-24%, the staking equilibrium for vault node operators could in fact be lower than this. However, for the purposes of the modelling, I will stick to an average of 20% for the staking equilibrium. In other words, in order to attract $7m worth of collateral to the vault nodes, the vault nodes should be earning enough Loki to have an annualized return rate of 20% where there is $7m of collateral in the vault nodes.

**So What Do The Numbers Say?**

The numbers really depend on how much Loki is worth at a given time. Obviously, price performance has been quite good lately, and there's probably still quite a bit of room to go. Unfortunately, it's hard to put a concrete answer on what will be sufficient and what will not be sufficient, due to the price volatility.

I've modelled this at a range of prices, all the way from 40 cents up to several dollars. A conservative estimate would have the best modelling conducted at the lower range of these boundaries. The final number that I have arrived at is that they will have to be awarded 10 Loki per block. For context, the miners currently on the network receive 22% of the emissions, which is 6 Loki per block.

If you want to read more about the modelling that I've done, see the spreadsheets, and play around with them yourself, I have [attached a workbook for you to look at](https://github.com/Haafingar/simon/blob/master/Blockswap%20Economics.xlsx). It accurately models the emission rate in the current scheme, and also looks at different price outcomes and different staking requirements for a set of 100 vault nodes.

**How Do We Do That?**

Proof-of-stake is coming up. Once the Pulse upgrade is in, which isn't that far away, miners will no longer receive 6 Loki per block. The obvious thing to do is to give that 6 Loki per block to the vault node operators. However, that 6 Loki will be insufficient to attract enough collateral to those vault nodes to provide the target liquidity that we want in most conservative price scenarios.

As such, we need to agree on the course of action that we are going to take to support Blockswap. We could either agree to increase the overall emission rate once vault nodes are required, or we can take some of the rewards from the existing Service Node network. Personally, I don't mind which way the community decides to do it, I think there are good arguments both ways. Creating new emissions will likely result in a greater overall lockup given that there are more rewards available, however, I also think the Service Node network can tolerate a slight reduction in rewards, especially considering many of the current Service Node operators are likely to switch to the larger node size with vault nodes anyway.

**A Further Idea**

From looking at development timelines, it doesn't look like vault nodes will launch before Pulse.  There's a further issue of seeding liquidity pools that I haven't yet discussed. I think that this is an important step in bootstrapping Blockswap to be a very successful product. While the block reward must attract liquidity to the vault nodes, there also should be a mechanism to attract liquidity to the vaults themselves. If there is no Loki or ETH to swap with, traders cannot use Blockswap. In order to encourage early provision of liquidity into these pools, I propose that a fund for pool seeding be created. This fund will be used to inject small amounts of Loki into the vaults, guaranteeing early liquidity providers at least some benefit from staking. The attraction of these rewards should create enough liquidity for traders to find it a viable option to swap assets across chains. Once the traders have appeared in sufficient volumes, seeding will no longer be necessary.

My proposition is that the 6 Loki per block that is currently being rewarded to miners, instead of being outwardly destroyed at the upcoming pulse hard fork, instead be directed to a Loki Foundation controlled wallet, and then be used to seed Blockswap liquidity pools and bootstrap the incentives for liquidity provision once the Blockswap team has launched the vaults. This will get the liquidity pools off the ground and make it much more likely that this product will be successful.

In order to limit this fund, I also propose that a timeout be built in to this redirection of the block reward. The hard fork that enables the vault nodes will remove this funding from the block reward, or if that hasn't occurred after 6 months, the fund will expire and no further Loki will be generated for this purpose.

**TL;DR**

At the Pulse upgrade, send the 6 Loki per block to a Blockswap liquidity fund (6 month expiry), then when the vault node upgrade comes out, remove the liquidity fund and either:
Add an additional 10 Loki per block to the emission schedule;
Cut 4 Loki per block from the Service Nodes and grant it and the current mining rewards to the vault nodes, or;
Some blend of the two.

**Conclusion**

Hopefully you’ve been able to gain some insight into the economics of Blockswap and can critically analyze this proposal with us. Please keep your feedback constructive and civil. Hopefully we can come to a consensus on the best way forward.

**Comment by SomethingGettingWrong** [comment]
My opinion would be 16.5 for the service nodes and 12.5 for the vault nodes would fix it based on my own calculations. I dont think we should lower the amount from the service nodes as that would break the social contract. However I feel there isnt enough from the 6 for vault nodes. I honestly also feel there isnt enough liquidity.  I think 29 loki per block would be a good ratio personally.

**Comment by Lucifer1903** [comment]
For now I'm going to agree with @SomethingGettingWrong

I also agree with diverting the 6 loki per block for 6 months into a Blockswap liquidity pools seed fund.

**Comment by Spootz** [comment]
I like the pool use idea, but I would say after the 6 months if the deadline hasnt been met and the pool still exists, any loki in the pool should be burnt out of circulation.

I somewhat agree with @SomethingGettingWrong, but I dont think those reward amounts are including the 5% loki foundation part of block reward, which I  think was meant to be changing to 10% after pulse? Is that still the case or not for this new proposed reward shake down?

So you would be looking at 29 loki emission plus loki teams 5-10% slice of the pie. 1 loki to dev is not needed when that is the whole point around the foundations portion of the block reward.

**Comment by lewisphil** [comment]
"If done in sufficiently large numbers, this will be able to offset the newly created coins awarded to the vault node operators."

If the goal is zero emissions added to the system by vault node operators, and their rewards are created at each block, couldn't the fee for each transaction be (the average of the last 5 blocks) per transaction?   

So if we knew there would be 100 transactions in the block that this transaction is included in, then we'd charge 6/100.  Since this cannot be known, but the last 5 blocks had 500 total transactions, we could charge 6/100 and programmatically (effectively) zero out the emissions.

If the emissions created are burned by code, then the SNops and LF ratios can be left, and changed by the normal process in the future.  In other words, they don't need to be accounted for in economics of the coin, since they are effectively nullified.

**Comment by SomethingGettingWrong** [comment]
> So you would be looking at 29 loki emission plus loki teams 5-10% slice of the pie. 1 loki to dev is not needed when that is the whole point around the foundations portion of the block reward.

yes, you are correct but i was more referring to at least 1 loki of the 5-10 slice  out of every block should be set aside and converted to USD everyday for a dev.. where as teh other portion of the loki foundations amount wouldnt be converted but instead held. I would like to have the FUNDS avaialble for at least 1 permenent deve.. EVEN though the loki foundations funding is out of hte block reward its kept as loki until they decide to spend it and get USD to pay for things. In this way at least 1 loki per block is a "known" and is always being converted and therefore there will always be dev funds. 

The only reason I mention it is that in the future  there will not be any more block reward changes and such a change needs to happen now.

**Comment by jagerman** [comment]
> If the goal is zero emissions added to the system by vault node operators, and their rewards are created at each block, couldn't the fee for each transaction be (the average of the last 5 blocks) per transaction?

Yes, although I think that the average of the last 5 is probably much too noisy: a rolling average over the last 1-2 days would probably work better.  There's also no reason that it needs to break even -- we could set a *negative* target so that `blockswap emissions = burn amount (averaged over the last 2 days) - 5`.  This would give some predictability, and offset coins emissions from the (regular) service node side.

But this is a longer term target that requires the system be operational and being used; much of the discussion here is about how to bootstrap that process.

**Comment by Haafingar** [comment]
You can't set a dynamic reward, at least not at the start. vaults have to stable, lasting at least 4 weeks at a time, and cost a fair bit to run. The fees burned at the time of swapping are going to vary greatly depending on prevailing market conditions and that volatility needs to be kept isolated from vault nodes. By giving them a fixed reward from the emissions schedule, we acheive exactly that. If more rewards are needed, we can decide on how and when to do that as a community.

We already receive 10% of the block reward as the foundation, and let me tell you, dumping an extra 1% would not help much at all. We've been pretty good at managing the funds so far. If your concern is that there isnt always going to be 'something' - there will be, but your proposal neglects to consider we're already over 20+ full time staff, and dumping 1% of the block reward doesnt even offset a single developer. We're much better off holding on the the Loki and selling it in bulk at a high price off the market, rather than providing this constant downward pressure.

**Comment by Spootz** [comment]
Yeah as much as I dislike the current emission rate of loki, and was hoping pulse would help control this a bit more by giving service node ops more to lock up, I dont think at this point in time if you want the blockswap side to work a negative target isnt feasible on initialisation. I am still with @SomethingGettingWrong method, minus the 1 loki. For the record loki foundation at 10% is fine as long as they are keeping it off the market.

Good to see a healthy discussion.

**Comment by jpthor** [comment]
All things here discussed are directionally correct. My position is:

1) Service Nodes receive payment for validating blockchain and servicing lokinet
2) Vault nodes receive payment for securing Blockswap
3) LPs receive payment for providing liquidity
4) LF receives long term funding support

Not having long-term incentives for LPs won't be competitive when stacked up to Ethereum DeFi, or THORChain, which pays 33% of incentives to LPs. 

Suggestion, if 100 units is the Block Reward:

Service Nodes: 40 units (they validate the blockchain - most important job)
Vault Nodes: 35 units (they secure assets - second most important)
LPs: 15 units (must pay LPs to be competitive and drive capital formation
LF: 10 units
Total: 100 units

**Comment by Lucifer1903** [comment]
I think it's important to be competitive with other projects that want to attract liquidity.
How do we do this and reduce the and burn enough loki to offset emissions?

**Comment by SomethingGettingWrong** [comment]
changing the block reward to 100 would frankly make me dump all my Loki to be truthful that is much to far from the original social contract.  **Whatever the number of the blockreward... each time we change it massively... it changes sybill resistance. servicenodes need the majority!!!!!!!!**

16.5 is what they are now. Forget the 1 loki per block i mentioend earlier for a permenent dev. 

IF the servicenode reward was 16.5 then valut nodes and everything else underneath it needs to be less then 16.5

**Comment by Lucifer1903** [comment]
@SomethingGettingWrong I think JP was using 100 as an example so people can see the percentages better in his explanation.

**Comment by SomethingGettingWrong** [comment]
> @SomethingGettingWrong I think JP was using 100 as an example so people can see the percentages better in his explanation.

Yes, but prior to that Simon said they already receive 10 percent through Loki foundation and that 1 percent wouldn't take care of a developer at all. That would only make since on a 100 block reward.  Prior to that i had suggested 1 Loki out of each block with the respect to the current block reward and at current prices that would be 720 Loki a day at 60 cents which is over 300 bucks a day.  So it makes me tend to think they already are looking at 100 as a block reward as that would make 1 Loki of the block reward equal 1 percent. Maybe he thought i meant 1 percent from the beginning? 


Either way I feel going to such a large block amount would cause all previous emitted loki  to lose value a the same ratio the block reward grows. 

Lets pretend for a second he did which would be fine.

The reason 100  reward would doesnt make since to me is because the social contract started at 45k and went down to 15k and the reward was 16.5. 

So cross dividing 40/16.5 to get a multiple.. it would equal 2.4242.  Then 2.4242* 15,000 = 36,363 service node requirement to keep the social contract the same. but then people that have say 4200 loki and were 1/4 a node cant stake... unless they allowed  ahigher multiple then 1/4   say 1/9 which would be 4,040.33. Then the person is only getting 1/9the the block reward instead of 1/4th.  . so 100 is not a good blockreward number.  


It needs to go up by a common factor of 2 which is the minimum stake requirement when determinign the block reward... .other wise the social contract is busted....


so double the amount 

33 for service nodes. 
24.87 for Vault Nodes
16.375 for Loki Pool 
8.25  for Loki foundation

82.495 block reward.    

This allows for 10 percent loki foundation. It allows for a multiple of the orignal service node social contract.  It allows for a good ratio for vault nodes and a large enough number i think for the loki pool.  While at the same time allowing service node operators to shrink the number of nodes that are too much too handle and are on the same Virtual private server to begin with by a factor of 2... which would also keep the cost ratio of purchasing power the same... since the nodes rewards would doulbe from 16.5 to 33 but the nodes themselves would go from 15k to 30k each at a minimum. This would also make rewards come 2x as often and then we change the stakign allowance from 1/4 to 1/8th which would naturally allow those that recieved a set amount for 1/4th an node at 16.5 to now recieve 1/8th at 33  which is 4.125 loki. 

16.5/4= 4.125
33/8= 4.125 

So basically in this way the block reward receives a different increment. but purchasing power is evenly divided from the old social contract.

**Comment by lewisphil** [comment]
"Even if that new lockup is being used to secure collateral in Blockswap, it still isn’t liquid — so it can’t be used in a Sybil attack on the network. "

If the attacker simply buys all the loki in all the vaults with BTC, does the price quoted rise exponentially as it would on an exchange?  Otherwise, how is this Loki still protecting the network?

**Comment by Lucifer1903** [comment]
I agree that we need incentives to attract liquidity.

I propose a new design for blockswap that will increase the security and incentivise liquidity providers to contribute.

Instead of creating a new type of node (vault nodes) we use the already existing service nodes to create Vault quorums. A vault quorum will consist of 10 service nodes and each service node can only be in one vault quorum at a time. These vault quorums will perform the functions of a single Vault nodes as already defined by the loki team and require a supermajority of 7 out of 10 service. To summarise each vault will consist of a quorum of Vault quorums and require a supermajority not only within each vault quorum but also within the quorum of Vault quorums (making them more secure) 

As service nodes already have a block reward designated to them this allows us to use the block reward that is planned for vault nodes to go to incentivising liquidity providers instead. 

If service nodes don't want to participate in vault quorums they can opt out for a reduced block reward for a reduced reward from 16.5 to 10.5.

Additionally we could also still have vault nodes alongside vault quorums (as I understand one of the suggested benefits of Vault node is that if large holders don't have to manage lots of service nodes they will be more likely to lock up their Loki too). If vault nodes are also implemented they could be limited to a maximum of 50% when creating a vault with the remaining vault consisting of Vault quorums. As Vault nodes don't perform all the functions service nodes do they could be paid less than service nodes. Let's say a vault node requires x10 the amount of loki that service nodes require, then they could be put in the reward queue 6 or 7 times, less work and easier management for a lesser reward.

**Comment by VINCEG-afk** [comment]
Whatever you do don't increase emissions. Cut sn rewards but do not increase emission.

**Comment by SomethingGettingWrong** [comment]
Block reward was originally 45.00  

Why not set the stake requirment on future service nodes to be 45,000 again. this keeps an incentive not to unstake 15,000 service nodes... are any currently staked servicenode...

Put block reward back at 45.00 in this way the increased emission per loki staked is the same and an increase blockreward doesnt result in an increase in emission on new service nodes. 


Block reward (22.50) for all new service nodes and old staked nodes that dont unstake. New requirment is 45,000 again. 

Remember orgianlly the 6.00 from miners was gonna go to the 16.5 we have now and that would equal 22.50 ANYWAY.. so the servicenode reward was gonna go to 22.50 there about..

Why not. 

22.5 for service nodes. 
10.00 for vault nodes
8 for Loki Pool
4.5 for loki foundation.

**Comment by Lucifer1903** [comment]
@SomethingGettingWrong how about using the service nodes we already have instead of creating a new type of node? The current mining fee can go into incentivising liquidity providers.

You're always saying how loki has a liquidity problem and this seems like it would solve that.

**Comment by lewisphil** [comment]
At current prices we have $28M MC, $7M is 25% of MC.  45% is locked up, so price goes up significantly and/or the node count drops significantly. 

Miners only account for 800K loki over 6 months, so there's a severe liquidity problem even when this hits mainstream, even if we're listed on all the major exchanges. 

I think there needs to be an incentive program for people to add to the liquidity pool during the six month funding startup window.  Maybe a reward bonus upon actual product launch. 
   i.e.  If you are part of the initial liquidity providers, 75% of the first 6 months of fees (or a guaranteed ROI) are paid to a fund that pays out proportionally to the initial providers.  SN operators can direct rewards to the fund / and / or designate some of their locked coins to the fund (since these coins still protect the network, according to the LRC.)

**Comment by lewisphil** [comment]
> @SomethingGettingWrong how about using the service nodes we already have instead of creating a new type of node? 

This feels more like a SNapp on top of the network of SNs.  I think severability of the vault nodes if this program doesn't work should be maintained.  Also allowing people who don't want to run a SN, but do want to run a vault node would attract more people. 
It could be another app on the same node, so SN operators could run it concurrently, or a vault node operator could get a vps for solely this purpose.  That would provide for the most flexibility.

**Comment by Haafingar** [comment]
> 
> 
> "Even if that new lockup is being used to secure collateral in Blockswap, it still isn’t liquid — so it can’t be used in a Sybil attack on the network. "
> 
> If the attacker simply buys all the loki in all the vaults with BTC, does the price quoted rise exponentially as it would on an exchange? Otherwise, how is this Loki still protecting the network?

The collateral staked into vault nodes is not the same as the liqudity in the pools. Collateral in the vault nodes *is* illiquid.

As for the second part of your question, yes, the fees that the buyer would pay rise exponentially as they consume more and more of one side of the pool's liqudity. 

As for @Lucifer1903 and the idea of using the existing service node network, there are some limitations on the way we want to do vaults that prevent 'multisig on multisig' operations that would make it possible for service nodes to sign vault transactions at a large enough scale to make it secure. Service nodes are too numerous, under collateralized, too low powered, and not reliable enough to do the work of a vault node. It makes much more sense to get the vault nodes operating as a seperate class of node, who have to be aware of multiple blockchains, operate a side chain, be extremely fast at MPC and peer communication, and have excellent reliability.

If the concern is that we're spending too much emission on SNs, then why not advocate for the SN rewards to be redirected to vault nodes?

**Comment by Lucifer1903** [comment]
@Haafingar my main issue that no one seems to be answering is; given the option of providing liquidity to blockswap vs another similar project that pays out part of their block reward, why would a liquidity provider put their liquidity in loki's blockswap when they can earn more money with a different project?
If you don't see having low liquidity as a problem then let me ask you, why would someone make an exchange on blockswap when it causes the price to swing and they pay more in fees than if they used another option? 
Sure privacy is a benefit, everyone knows session needs to be free or the majority of people won't use it, blockswap need to be competitive in fees with other projects or the majority of people will go elsewhere.

I rambled on a bit. Basically please explain why would a liquidity provider choose blockswap when they earn less? And why would a user exchange coins on blockswap when it costs more in slippage?

**Comment by Haafingar** [comment]
I never said I thought liquidity wasnt a concern. This is why I suggest we use the 6 Loki per block to create a seed fund to create early incetives for liqudity provision. 

However, the system has got to stand on its own two feet. Systems that rely on printing tokens to give to liquidity providers are essentially funding the system at the expense of token price. COMP, BAL, etc are going to zero on a long enough time frame.

Blockswap won't have to do this long term if the product is good and people are actually using it to swap coins. Its the only solution that allows for generic cross-chain swaps with no special wallets - if we had to permentantly pay LPs, that to me says we don't have a unique selling point. LPs stake to earn trading fees - that's the whole premise of AMM. Trading fees will come through Blockswap because no other pool of this type exists for swapping between BTC, ETH, Polkadot, etc seamlessly. There's no point for competing over USDT/ETH pools for example, we'll never win that liqudity, we're better off just hooking into uniswap for that.

**Comment by Skelaton4** [comment]
At the Pulse upgrade, send the 6 Loki per block to a Blockswap liquidity fund (6 month expiry), then when the vault node upgrade comes out, remove the liquidity fund and either:

Cut 4 Loki per block from the Service Nodes and grant it and the current mining rewards to the vault nodes, 

I favor this.

**Comment by yidakee** [comment]
I hope some anti-dump mechanism is introduced. Say for example 250k Loki get slashed for some odd reason, instead of doing the typical fee payout to a specific "miner", dilute it so the entire network benefits and less of a chance of a market dump

For ex: match 1:1 block rewards until the slashed Loki are consumed, or similar mechanism.

**Comment by Lucifer1903** [comment]
I think reducing the service node block reward to make up for the vault node reward is the best option. A lower reward for service nodes will incentivise operators to move to vault nodes. As operators move from SNs to VNs the SN reward will will be split between less nodes so the profitably of SNs shouldn't change too much.

**Comment by Spootz** [comment]
I've felt like for a long time that the emissions are too great and the market has to swallow the emissions limiting the value of the coin. Loki has great features already that are very marketable. The tech team has done exceedingly well in generating these products. I feel that these new proposals wouldnt be so scary in the way of inflation if the token had a price that reflected the progress achieved. At some point we as a community are going to have to agree to take a risk in which we wont be relying on the ability to mass print tokens to incentivise our service providers. If the value of the coin is great, you rely less and less on having to provide large emissions to achieve these services.

My point is, let's also not get too caught up in the new stuff and forget about what we have already achieved. Market the token and you wont have to run numbers on the lower end between 40c to $7.

**Comment by Haafingar** [comment]
I'm not sure why so many people are talking about what happens to slashed funds. The answer is easy: They get burned. Ideally no one ever gets slashed, it should only be reserved for activley malicious nodes, and the existence of the penalty should prevent it from ever having to be used.

Ineteresting to see the consensus leaning towards cutting SN rewards to fund VNs!

**Comment by Lucifer1903** [comment]
@Haafingar would it make sense for VNs to not disclose IP address? If a hacker knows all the IP address for the VNs that are making up a Vault then they know exactly which IP address they need to attack to either gain funds or DDoS a Vault.
It would be good if they were hidden behind lokinet and when it's time to create a new vault the nodes selected for said vault connect to each other over lokinet and share their real IP address for a faster connection.

**Comment by Haafingar** [comment]
VNs very likely will have their IPs exposed just as SNs do. VNs will need to take their security more seriously than a typical SN, but your run-of-the-mill debian server isn't that easy to crack if set up using best practices. Routing VNs through Lokinet would cause a significant slowdown to vault signing ceremonies which involve 7 rounds of communication using GG20. Latency needs to be kept to a minimum.

DDoS is a legitmate problem until you consider that VNs only have to talk to Quoters and SNs, both of which have their IPs registered on the blockchain. Effectivley, VNs can easily firewall themselves off from anyone not whitelisted on the Loki blockchain.

**Comment by Lucifer1903** [comment]
@Haafingar can't IP address be shared between Vnode only after they have been selected into a quorum? After they privately share their IP address they can connect to each other directly for all vault quorum tasks such as signing ceremonies without any slowdown.

**Comment by Haafingar** [comment]
VNs very likely will have their IPs exposed just as SNs do. VNs will need to take their security more seriously than a typical SN, but your run-of-the-mill debian server isn't that easy to crack if set up using best practices. Routing VNs through Lokinet would cause a significant slowdown to vault signing ceremonies which involve 7 rounds of communication using GG20. Latency needs to be kept to a minimum.

DDoS is a legitmate problem until you consider that VNs only have to talk to Quoters and SNs, both of which have their IPs registered on the blockchain. Effectivley, VNs can easily firewall themselves off from anyone not whitelisted on the Loki blockchain.

**Comment by jagerman** [comment]
> Routing VNs through Lokinet would cause a significant slowdown to vault signing ceremonies which involve 7 rounds of communication using GG20. Latency needs to be kept to a minimum.

One potential idea here is that the only public info would a .loki address, then VNs would connect to each other there, authenticate that they are talking to a valid vault node, then exchange public IPs and reconnect through that.  Then you get the low latency of a direct IP connection, but don't have to expose the IP address publicly.

**Comment by Lucifer1903** [comment]
@jagerman thank you for saying what I wanted to with better words haha

**Comment by SomethingGettingWrong** [comment]
The other day I was extremely "wordy" and not very clear in what i was trying to say the other day. I have finished the necessery data to show easier what I was trying to say. 

While the number Service Nodes are Vault Nodes are un-kown what is known is that they will find balance on ROI. The service nodes will un-stake into vault nodes and the vault nodes will un-stake into service nodes until ROI is equal on the chosen block rewards. 

I have created a calculator to input the various staking requirements and block rewards along with inputs to an unknown quorum size and penalty slash amount and filled them with 80 percent and 30 for quorm. Although these dont have anything to do with the high lighted calculations. 

The point I was trying to easily say the other day is that the confusion was that I was saying the block reward wouldn't balance out are that it was somehow wrong. Any block reward and node count is gonna find balance. I have a calculation for finding balance depending on the nodes entered at the various settings. At 100 percent the Emission for each individually staked LOKI in both nodes ROI is equal! This is where the free market will most likely land. The nodes will fill until they find that equilibrium. 

The problem isn't the amounts but the amounts relative to one another that I have issue with. While they can find balance and as you will see i will put in different node counts and balance them to 100 percent.  The ROI will change. You will also notice "The vault nodes can purchase a service node and a service node can purchase a vault node calculations."

Below is a few sample images. To point out something I believe is being overlooked. Sybil Resistance.  How long before the vault node owners can buy all the service nodes faster then the service nodes can buy vault nodes matters based on input!

While this is not an issue if New money comes in I just merely want to point out if its the same node owners un-staking and buying vault nodes. Then these vault nodes continuously stake they will eventually overtake the service node owners emission if they choose to only run a service node. I get this is an amazing and ample opportunity. I just wanted to input a few examples to show the variance of years. Its not the block reward number itself that matters as to me as much as the time frame we have before approaching 51 percent of vault node owners taking all their emission and re-staking into service-nodes faster then old service-node owners can re*stake into service nodes. 

**At 800 service Nodes at 16.5 block reward and 111 vault nodes at 10 block reward the vault nodes will be able to re-stake all rewards and own over 50 percent of the service nodes in  4.75 years with ROI being 4.376 years on vault nodes and servicenodes at 100.40% equal emission per loki staked.**  

![image](https://user-images.githubusercontent.com/36722911/89839224-e11f6400-db32-11ea-89ee-06bef18a22c9.png)

**At 800 service Nodes at 12.5 block reward and 146 vault nodes at 10 block reward the vault nodes will be able to re-stake all rewards and own over 50 percent of the service nodes in  2.74 years with ROI being 4.376 years on vault nodes and servicenodes at 100.05% equal emission per loki staked.**  

![image](https://user-images.githubusercontent.com/36722911/89839700-08c2fc00-db34-11ea-9cd1-f75bf0d89553.png)

I dont think 2.74 years is enough time. 

thats why i think we should start out at 6 blockreward and increment up to 10 block reward over 2.74 years instead! That way we can achieve the 10 block reward while at the same time not overpaying new nodes! Just look at if there was 1 node online the entire 7200 emission (720blocks*10breward) would be his.

![image](https://user-images.githubusercontent.com/36722911/89839914-7e2ecc80-db34-11ea-9837-a853253e3096.png)




basicly I think 12.5 and 100 will be okay.. but we can extend the time period by starting the block rewards at 16.5 on servicenodes and counting down over a period of time to 12.5 and starting the vault nodes out at 6 block reward and counting up to 10 block reward over that same period of time per block over the time period the team thinks the vault nodes and loki nodes will take to balance! 3 months?

**Comment by Haafingar** [comment]
Thanks for that Patrick. Glad to see that you've made some adjustments.

You're saying that Sybil resistance has been overlooked, but as long as we assume that the initial set of Service Node operators are reasonably diverse (which they are) and vault nodes follow a similar even distribution between honest actors, the scenario you are describing will never happen. There will always be some level of new money entering and old money leaving, and the situation you are describing would imply that all Service node and vault node operators are colluding and trying to Sybil attack the other.  _It isn't a realistic scenario._

I have argued, **and maintain**, that adding Vault nodes **increases** Sybil resistance from a malicious actor that doesn't already have a large stake in SNs. The VNs will cause more of the circulating supply to be locked up, making it even more expensive for new actors to arrive and attempt to take over the network. 

If you believe current operators aren't honest and the Loki network is corrupt, then that would be a different story. I think we can all agree that the distribution of operators is currently fine, and given that fact, **the addition of VNs will only serve to bolster the entire network** (both blockswap vaults and PoS/Session/Lokinet) and its safety against Sybil attacks.

**Comment by SomethingGettingWrong** [comment]
The problem is  the system is setup to incentive all vault nodes to put rewards into staked service nodes so they can gather those as well. vault nodes always receive more then service nodes with same ROI per staked loki... then their is no need for bad actors.. and no need for old vault nodes to  restake into vault nodes since they would just be competing against vault node reward competition agains thier own nodes only the richest can purchase a full vault node. 

Everyone will do what makes them profit the most which would be to stake vault node emissions into service nodes. which means that the first few vault node owners that control majority of emission now will give up their position of having a larger number of nodes then most people as they unstake to stake into vault nodes which will allow enough emission to slowly reclaim thier postion with a larger number of nodes back into service nodes again. 

This is just how it is I get that. Im just hoping the nodes come online Extremely fast otherwise we are going to have an opportunity for new money to actually buy into vault nodes and then later control service nodes even though the total amount sybill resistance goes up with each staked loki.

Im hoping their can be partial node stakes like on service nodes? This will allow old money of good actors thats been around since the beginning of the network to actually be able to buy into the vault nodes.

**Comment by Haafingar** [comment]
Last calls for comments on LRC 7 before a conclusion is drawn! I'll leave it open for another few days just in case anyone else wants to weigh in after they've had a chance to review the full Blockswap paper released by team

**Comment by Haafingar** [comment]
Conclusion:

At the Pulse upgrade,  redirect the 6 Loki per block mining reward to a Blockswap liquidity fund (6 month expiry). If the liquidity fund reaches it's expiry time, the network will simply stop producing it and scale back LF rewards so that it is still 10% of the reward. 

At the next upgrade that includes Vault Nodes, this 6 Loki per block can be combined with a 4 Loki per block cut from the SN reward pool and will be granted to Vault nodes.

We may open a new LRC before that upgrade as a confirmatory exercise in this token economics plan.

BRING ON PULSE!
